src/client/Microsoft.Identity.Client/ApiConfig/BrokerOptions.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Microsoft.Identity.Client.ApiConfig
+{
+    /// <summary>
+    /// The class specifies the options for broker across OperatingSystems
+    /// The common properties are direct members
+    /// Platform specific properties (if they exist) are part of the corresponding options
+    /// </summary>
+    public class BrokerOptions
+    {
+        /// <summary>
+        /// Supported OperatingSystems
+        /// </summary>
+        [Flags]
+        public enum OperatingSystems
+        {
+            /// <summary>
+            /// No OS specified - Invalid options
+            /// </summary>
+            None = 0b_0000_0000,  // 0
+            /// <summary>
+            /// Use broker on Windows OS
+            /// </summary>
+            Windows = 0b_0000_0001,  // 1
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="osChoices">Choices of OperatingSystems</param>
+        /// <param name="title">Title of the broker</param>
+        /// <param name="msaPassthrough">is this MsaPasstorugh</param>
+        public BrokerOptions(OperatingSystems osChoices, string title = "", bool msaPassthrough = false)
+        {
+            if (osChoices == OperatingSystems.None)
+            {
+                throw new ArgumentException($"osChoices should not be none");
+            }
+
+            OSChoices = osChoices;
+            Title = title;
+            MsaPassthrough = msaPassthrough;
+        }
```

Comment:
I think this constructor should just accept OS enum and properties should be set by themselves because:
- If add new properties later, we'd have to add them to the signature every time.
- Updating the method signature with optional params is a binary breaking change and if we make params required, we'll end up with many overloads.
- The properties are public anyway.

I think this is more simplistic use, users can set options they care about.
```suggestion
        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="operatingSystems">Choices of OperatingSystems</param>
        public BrokerOptions(OperatingSystems operatingSystems)
        {
            if (operatingSystems == OperatingSystems.None)
            {
                throw new ArgumentException($"Operating system must be specified.");
            }
        }
```